extensions/pkg/bastion/vmdetails.go

```diff
@@ -0,0 +1,217 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bastion
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/exp/maps"
+	"k8s.io/utils/ptr"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+)
+
+// VMDetails define all bastion vm details derived from the CloudProfile
+type VMDetails struct {
+	MachineName   string
+	Architecture  string
+	ImageBaseName string
+	ImageVersion  string
+}
+
+// DetermineVmDetails determines the bastion vm details based on information in the cloud profile
+func DetermineVmDetails(spec gardencorev1beta1.CloudProfileSpec) (vm VMDetails, err error) {
+	imageArchs, err := getArchitectures(spec.Bastion, spec.MachineImages)
```

Comment:
Out of curiosity:

Would this step be necessary if there is machineType defined in the bastion config ? Does it make sense to have to only execute the `getArchitectures` when a machine is not defined ?

Reply:
That is true.
I changed the logic as following:
```
if bastionSpec != nil && bastionSpec.MachineType != nil {
	vm.MachineTypeName, vm.Architecture, err = getMachine(bastionSpec, profile.Spec.MachineTypes)
	if err != nil {
		return MachineSpec{}, err
	}
} else {
	vm.MachineTypeName, vm.Architecture, err = findMostSuitableMachineType(profile)
	if err != nil {
		return MachineSpec{}, err
	}
}
```